scripts/diagnostic_gen.py

```diff
@@ -8,15 +8,14 @@
 import os
 import shlex
 import subprocess
+import sys
 
 
 def writefile(path, contents):
-    try:
```

Comment:
Why this change to remove the try block?

Reply:
As I was debugging the script for the hermetic build I looked at the code in this function.

I am of the old school that believes we should use error checking when available instead of trying to handle errors with an exception. I know it is a matter of taste, but in this case, where the exception doesn't even propagate, I find the code cleaner. Happy to defer to you though, if you prefer the try/catch, I can revert this change. I will still continue to argue that the code is clearer with the check :)

Reply:
That's generally a good approach for a language like C++ but in Python it's considered idiomatic to use the exception path. It's also in this case not a correct change; doing `if (exists(file)) { open(file); }` has a race condition (in any language) where the file can be deleted in between the two calls. Doing it exceptionless requires an `open` method that returns an error code rather than throws.

Reply:
Reverted it.